packages/cli/commands/project/deploy.js

```diff
@@ -59,27 +79,46 @@ exports.handler = async options => {
   let buildIdToDeploy = buildIdOption;
 
   try {
-    if (!buildIdOption) {
-      const { latestBuild, deployedBuildId } = await fetchProject(
-        accountId,
+    const { latestBuild, deployedBuildId } = await fetchProject(
+      accountId,
+      projectName
+    );
+
+    if (!latestBuild || !latestBuild.buildId) {
+      logger.error(i18n(`${i18nKey}.errors.noBuilds`));
+      return process.exit(EXIT_CODES.ERROR);
+    }
+
+    if (buildIdToDeploy) {
+      const validationResult = validateBuildId(
+        buildIdToDeploy,
+        deployedBuildId,
+        latestBuild.buildId,
         projectName
       );
-      if (!latestBuild || !latestBuild.buildId) {
-        logger.error(i18n(`${i18nKey}.errors.noBuilds`));
-        process.exit(EXIT_CODES.ERROR);
+      if (validationResult !== true) {
+        logger.error(validationResult);
+        return process.exit(EXIT_CODES.ERROR);
       }
+    } else {
       const buildIdPromptResponse = await buildIdPrompt(
         latestBuild.buildId,
         deployedBuildId,
-        projectName
+        projectName,
+        buildId =>
+          validateBuildId(
+            buildId,
+            latestBuild.buildId,
+            deployedBuildId,
+            projectName
+          )
       );
-
       buildIdToDeploy = buildIdPromptResponse.buildId;
     }
 
     if (!buildIdToDeploy) {
       logger.error(i18n(`${i18nKey}.errors.noBuildId`));
-      process.exit(EXIT_CODES.ERROR);
+      return process.exit(EXIT_CODES.ERROR);
```

Comment:
These `return process.exit(...)` are a hack to improve the testability of this method because these calls were killing the test process.  Since the `process.exit` method will kill the process so we don't actually `return` anything in a non test environment.  But when these methods are mocked, we are able to test only up until `process.exit` method is called.  I'm not crazy about adding code just for the sake of testing, but the other options I could come up with were to mock `process.exit` to throw an exception, which then triggered our `catch` block, or to mock `process.exit` to just return null, which meant the rest of the method just kept running.  

This felt like the lesser evil

Reply:
Works for me! Don't see any downside to returning when everything is exiting anyway